onnx/reference/op_run.py

```diff
@@ -403,6 +403,40 @@ def _run(self, *args, **kwargs):  # type: ignore
             f"Method '_run' must be overwritten for operator {self.__class__.__name__!r}."
         )
 
+    def _check_output_run(self, res: tuple[Any]) -> tuple[Any]:
+        """
+        Checks the output are from the expected type.
+        """
+        if not isinstance(res, tuple):
+            raise TypeError(
+                f"Method '_run' of class {self.__class__.__name__!r} does not return a tuple but {type(res)}."
+            )
+        if len(res) == 0:
```

Comment:
```suggestion
        if not res:
```

Reply:
I was wondering why this was a recommandation.

```python
import timeit
print(timeit.timeit('if len(A) == 0:\n  pass', setup='A=[1,2,3]'))
print(timeit.timeit('if A:\n  pass', setup='A=[1,2,3]'))
```

It turns out it is 4 times faster. I still wonder why.